### ISSUE TITLE
Expose OAuth class through Auth module

### ISSUE THREAD

**Issue by jglesner** [aws-amplify/amplify-js#8933]
### Describe the feature you'd like to request

In issue #8632, I highlighted the fact that there is limited documentation/support for programmatically create a user session using Cognito (id, access, refresh) tokens.  In the [solution I provided to my own question](https://github.com/aws-amplify/amplify-js/issues/8632#issuecomment-886133768), I am using a private method in the Auth module: _oAuthHandler.  My approach bypasses the compile time checking, since this method is marked private, which isn't a good long-term solution.  

I use that method because it exposes the OAuth class, which has the handleAuthResponse method to conveniently process a cognito response url.  Therefore, I would like to see the OAuth class (or at least the handleAuthResponse method in OAuth) made public.  While its possible for someone to write their own RESTful method to process the cognito response url, what's the point of doing that when the AWS Amplify package exists?  I need to process the cognito response url directly, and not rely on the `urlListener` in my Cordova app because there is limited support for deep linking. 

### Describe the solution you'd like

I would like _oAuthHandler method to be made public so that I can access OAuth class without having to bypass the fact that it the _oAuthHandler is currently private, and so that I/others know that this method will be supported into the future. 

If _oAuthHandler is made public, you may want to drop the _ at the front.  And I would use it like so:  

```javascript
import * as AmazonCognitoIdentity from "amazon-cognito-identity-js";

// passing a valid cognito response url with authorization_code and state
Auth.oAuthHandler.handleAuthResponse(url).then((resp) => {

    // using the id, access and refresh tokens provided by the AuthHandler:
    const AccessToken = new AmazonCognitoIdentity.CognitoAccessToken({
           AccessToken: resp.accessToken,
    });
                
    const IdToken = new AmazonCognitoIdentity.CognitoIdToken({
          IdToken: resp.idToken,
    });

    const RefreshToken = new AmazonCognitoIdentity.CognitoRefreshToken({
           RefreshToken: resp.refreshToken,
    });

    const sessionData = {
          IdToken: IdToken,
          AccessToken: AccessToken,
          RefreshToken: RefreshToken,
    };

    const session = new AmazonCognitoIdentity.CognitoUserSession(
          sessionData
    );
});
```

### Describe alternatives you've considered

The alternative is to make a REST call to the Cognito API, or bypass compile time checking, which isn't a good solution. 

### Is this something that you'd be interested in working on?

- [X] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

**Comment by cxonexus-github-admin** [comment]
+1 Setting up SAML Federated access and am working on this exact issue right now.

**Comment by stemwinder** [comment]
+1 SAML federated scenarios

**Comment by bobthebuilder1997** [comment]
+1 SAML federated scenarios

**Comment by chanpod** [comment]
Need for some micro UI SSR scenario's (so basically a federated setup I think)

**Comment by abdallahshaban557** [comment]
Hi @jglesner  - I want to make sure I am understanding your use case properly - you are using federatedSignIn, but then need to create a user session which produces the id, access, refresh token sessions programmatically as part of that flow?

**Comment by jglesner** [comment]
@abdallahshaban557 In my particular case, I'm using federatedSignIn in a different browser window in a JS mobile app, and once complete, I have the Cognito URL.  I want public access to the Auth class so that I can process it.  In my app, sign-in occurs in a separate popup-window, so I need a programmatic alternative to `urlListener`.  If `Auth.oAuthHandler.handleAuthResponse` were public, I can achieve this.  

Using the `handleAuthResponse` is a means to an end -- ultimately I want to create a cognito session recognized by Amplify.  One way to do that, is to use the id, access, refresh tokens I receive from `handleAuthResponse` to create a valid Amplify session, as I demonstrate in [#8632](https://github.com/aws-amplify/amplify-js/issues/8632).

**Comment by abdallahshaban557** [comment]
Hi @jglesner  -and for your particular use case using the AWS credentials is not enough to access resources you might need? you need to create a session and retrieve all the tokens you get with userpool sign in?

**Comment by jglesner** [comment]
@abdallahshaban557 no, this isn’t so I can log in. this is a mobile app backed by cognito designed to let users of the app create accounts. As I explained, I don’t need the tokens as much as I need to process a cognito url and create an Amplify cognito user session.

**Comment by saconnolly** [comment]
@abdallahshaban557 Has there been any update on this? From what I can see, @jglesner proposed solution is the best option as of now and none of us love using a private function.  

In my opinion, the best option would be leaving Amplify out of the solution and adding `handleAuthResponse` to the amazon-cognito-identity-js package.  Currently the amazon-cognito-identity-js doesn't have a documented solution for how to support Authorization Code Flow for OAuth and this seems like the best way to start

**Comment by abdallahshaban557** [comment]
@saconnolly  - we do not have an update yet. Thank you for that suggestion, however we are not making update to the `amazon-cognito-identity-js` package in the near future.
